### ISSUE TITLE
Bio and Signature infinite new lines...

### ISSUE THREAD

**Issue by effone** [mybb/mybb#3531]
Salam Alekum...

One of my members just reported something which interested me:
When you are writing in signarture box which show in your posts or your profile. You have 250 I think as limited characters. 
The error is that you can enter or going on new line 250 times, IF you do that the mybb system will generate more than that \b in the page for example:
I did tried 120 times entering the forum generated 5498 enters....

This is true for the bio box too...

https://cdn.discordapp.com/attachments/4...nknown.png

https://cdn.discordapp.com/attachments/4...nknown.png

My English not good, Didn't know what to chose as the title.
Thank you...

Original thread: [Bio and Signature infinite new lines...](https://community.mybb.com/thread-221100.html)

**Comment by Eldenroot** [comment]
So how to fix that? The easiest way is to restrict the max height of the signature box and hide scroll bar via CSS.

**Comment by effone** [comment]
Or .... wiping 2 or more blank lines at a stretch.???

**Comment by WildcardSearch** [comment]
I think this is easier to just fix with CSS by limiting the height of the signature.

But if you are bound and determined to fix it, you would have to replace newlines with a placeholder, get a count, check the length, and then proceed with error/success.

**Comment by effone** [comment]
Then there no issue remains. Users can do that at their own level.
Its really a weird situation.

**Comment by Ben-MyBB** [comment]
For this issue, we could suggest some CSS to be added for the fix but I know @justinsoltesz is against further CSS changes so I'm not sure this is even worth fixing in 1.8?

**Comment by euantorano** [comment]
The problem is that CSS changes don't work with the upgrade system - we have to rely on users making the changes manually and that always causes support tickets.

**Comment by Ben-MyBB** [comment]
We might be as well resolving this in CSS and leave it on this ticket or in a thread in the forums for people who are specifcally experiencing this issue to apply the fix? But given how long 1.8 has been out for and how long this issue has been open I don't think there's loads of people with this problem.

**Comment by justinsoltesz** [comment]
I’m not sure this is something we need to fix...at a certain point, you just have to moderate your forums. We provide tools for moderators to change or remove users’ signatures.

A maximum height can easily be set in the CSS in 1.9 (edit: for those who want to - I don’t think we should set a height by default)

**Comment by effone** [comment]
What CSS can save us here?
`br:not(:first-child){display: none;}` ? That looks dirty to me.

There are unwanted line breaks getting added and this should be fixed in js level, specially when we have forked uncle sam now.

**Comment by justinsoltesz** [comment]
You can just set a maximum height for the signature.

Like I said though I don’t see why we need to fix this. Someone could also make a post with 1000 line breaks. That’s a moderation issue, not a bug.

**Comment by effone** [comment]
I am sorry, but may be you are missing the point of the issue.
Its not about height of the signature (atleast not only). Thing is if user enters multiple blank lines in signature, the system is generating multiple blank lines, way way more than what the user has entered. For ready example from the reporter, he entered 120 blank lines and system returned him 5498, not even 120.

**Comment by euantorano** [comment]
@effone That sounds like an SCEditor bug that I have seen before, where sometimes it doesn't create any blank lines, and sometimes it creates way more than you entered. Unfortunately the screenshots are not loading for me to see if that is the same error I am thinking of.

**Comment by effone** [comment]
Yes thats what I was trying to say, its an SCE bug. I tried and reproduced successfully when I pushed the issue.

**Comment by euantorano** [comment]
Thanks for the clarification. We now have an SCEditor fork at [mybb/SCEditor](https://github.com/mybb/SCEditor) - would probably best to try and fix this there.
